### ISSUE TITLE
IndexStatsIT.throttleStats failure: index throttling didn't kick in 

### ISSUE THREAD

**Issue by bleskes** [elastic/elasticsearch#15037]
See http://build-us-00.elastic.co/job/es_core_21_metal/265/testReport/junit/org.elasticsearch.indices.stats/IndexStatsIT/throttleStats/

```
java.lang.AssertionError: index throttling didn't kick in after 5 minutes of intense merging
	at __randomizedtesting.SeedInfo.seed([1EEE4D54B95D30BC:A3ED42A92929FD08]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.elasticsearch.indices.stats.IndexStatsIT.throttleStats(IndexStatsIT.java:352)
```

Sadly, it doesn't reproduce. @mikemccand this is classically your baby. Feel free to reassign.

**Comment by mikemccand** [comment]
LOL my baby :)  I will dig ...

**Comment by mikemccand** [comment]
This looks fixed now by e3628ecad244f15f74badca63146d548d0869dd4 where I tried switching to async translog, since we are doing individual indexing ops.

It's odd this was necessary only on 2.1.x though, I can't explain that.

But I'll port forward to 2.2 and 3.0 as well.
